ntbs-service/Models/TreatmentPeriod.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using ntbs_service.Models.Entities;
+
+namespace ntbs_service.Models
+{
+    [NotMapped]
+    public class TreatmentPeriod
+    {
+        public List<TreatmentEvent> TreatmentEvents { get; set; }
+        public int PeriodNumber { get; set; }
+        public bool IsTransfer { get; set; }
```

Comment:
I think it would be a good idea to make these properties get only, since none of them should ever change. The `PeriodNumber` and `IsTransfer` should then be set in the constructor, and the `TreatmentEvents` should begin as an empty list which can then be added to. For extra safety, you could also have a separate constructor for transfer periods.

I also think that `PeriodNumber` would be better as a nullable int, because it would be better as `null` rather than `0` for transfer periods.

Reply:
I've refactored this roughly along those lines, because there are two ways to make this treatment period (either one with multiple events, or a transfer period), I've made two different methods to construct these objects, `CreateTreatmentPeriod` and `CreateTransferPeriod`. However, this has made me wonder whether these should be split into two classes, implementing some `TreatmentPeriod` interface. What do you think? Given how small the class is it might be over-engineering it though

Reply:
I think both approaches would be fine, but like you I'd probably fall on the side of saying that having separate classes was over-engineering in this case.